src/clusterfuzz/_internal/tests/core/swarming/swarming_test.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Swarming tests."""
+import base64
+import unittest
+
+from clusterfuzz._internal import swarming
+from clusterfuzz._internal.datastore import data_types
+from clusterfuzz._internal.protos import swarming_pb2
+from clusterfuzz._internal.tests.test_libs import helpers
+from clusterfuzz._internal.tests.test_libs import test_utils
+
+
+@test_utils.with_cloud_emulators('datastore')
+class GetSpecFromConfigTest(unittest.TestCase):
+  """Tests for get_spec_from_config."""
+
+  def setUp(self):
+    self.maxDiff = None
+    helpers.patch(self, ['clusterfuzz._internal.swarming._get_task_name'])
```

Comment:
Instead of mocking out a private helper, can we instead do:

```python
spec = _get_new_task_spec(...)
self.assertRegex(spec.name, 't-[a-z0-9\-]*')  # Or even just "starts with `t-`"

spec.name = ''
self.assertEquals(spec, expected_spec)  # Where `expected_spec` has an empty name
```

Reply:
it ended up being convenient for the `push_swarming_task` test so I had to keep it